### PR TITLE
[WIP]:Feat/callback

### PR DESCRIPTION
**Pull request by sunnycyk**
experimental code

### CODE REVIEW

File: __tests__/test.js [hunk 1]
```diff
@@ -75,6 +79,23 @@ describe('Hybrid-Cache', () => {
...
 
+    it('invalidate and update for every cache', async () => {
+      const Value2 = {
+        rich2: 'object2'
+      }
+      Assert.deepStrictEqual(cache.put(Key, Value), Value)
+      let cache2 = await Hybrid.Create(Topic, new MockPubSub(backingStore))
+      const p = new Promise( (resolve) => {
+        cache2.on('invalidate', (channel, key) => {
+          resolve(cache2.put(key, Value2))
```
Comment (lionello) [inline __tests__/test.js]: Add asserts for `channel` and `key`

Comment (lionello) [reply __tests__/test.js]: No need to `cache2.put` inside the `on`

File: __tests__/test.js [hunk 2]
```diff
@@ -75,6 +79,23 @@ describe('Hybrid-Cache', () => {
...
+        })
+      })
+      await cache.invalidate(Key)
+      const r = await p
+      Assert.deepStrictEqual(r, Value2)
+      Assert.deepStrictEqual(cache.get(r), null)
```
Comment (lionello) [inline __tests__/test.js]: r=>`Key`

File: __tests__/test.js [hunk 3]
```diff
@@ -75,6 +79,23 @@ describe('Hybrid-Cache', () => {
...
+      Assert.deepStrictEqual(cache.put(Key, Value), Value)
+      let cache2 = await Hybrid.Create(Topic, new MockPubSub(backingStore))
+      const p = new Promise( (resolve) => {
+        cache2.on('invalidate', (channel, key) => {
+          resolve(cache2.put(key, Value2))
+        })
+      })
+      await cache.invalidate(Key)
+      const r = await p
+      Assert.deepStrictEqual(r, Value2)
```
Comment (lionello) [inline __tests__/test.js]: remove this line

File: __tests__/test.js (hunk 1 above)
Comment (lionello) [reply __tests__/test.js]: Add assert for `cache2.get(Key)`: should it be the value we `put` before we invalidate

File: __tests__/test.js [hunk 4]
```diff
@@ -75,6 +79,23 @@ describe('Hybrid-Cache', () => {
       Assert.strictEqual(cache2.get(Key), null)
     })
 
+    it('invalidate and update for every cache', async () => {
+      const Value2 = {
+        rich2: 'object2'
+      }
+      Assert.deepStrictEqual(cache.put(Key, Value), Value)
+      let cache2 = await Hybrid.Create(Topic, new MockPubSub(backingStore))
+      const p = new Promise( (resolve) => {
```
Comment (lionello) [inline __tests__/test.js]: rename `p` to `promise`

File: __tests__/test.js [hunk 5]
```diff
@@ -75,6 +79,23 @@ describe('Hybrid-Cache', () => {
...
+        cache2.on('invalidate', (channel, key) => {
+          resolve(cache2.put(key, Value2))
+        })
+      })
+      await cache.invalidate(Key)
+      const r = await p
```
Comment (lionello) [inline __tests__/test.js]: no need for `r`

File: index.js [hunk 6]
```diff
@@ -103,6 +106,18 @@ Cache.prototype.invalidate = function (key) {
   return this._redis().publish(this.topic, key)
 }
 
+
+/**
+ * @param {string} eventName Only 'invalidate' is accepted
+ * @param {function} cb callback
```
Comment (lionello) [inline index.js]: also supports `null`: `{function|null}`

**Review by lionello — CHANGES_REQUESTED** [review]

**Comment by sunnycyk** [comment]
@lionello merge if ok! Thanks

**Review by lionello — APPROVED** [review]

**Pull request merged**